### ISSUE TITLE
Refactor and add section on README on how the validation and test set are handled

### ISSUE THREAD

**Issue by RomainGoussault** [SubstraFoundation/distributed-learning-contributivity#168]
- How do we use the validation data set ? For early stopping only ? Is it local or global ?

- Do we have a local or global test set ?

- etc

**Comment by RomainGoussault** [comment]
Looks like val_data is actually a subset of the train set, so it's a local validation_data ? https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/e5ca82f641b32aa7d61199875b09c8fa0a099864/dataset.py#L38

**Comment by bowni** [comment]
I confirm that currently the partner-level validation and test datasets are not used.

**Comment by RomainGoussault** [comment]
Proposition of the use of local/global validation and test sets (I will not discussion implementation details here). 

- **locally** (on each partner node), we don't need both a validation and a test set. We only need one dataset to locally evaluate the model. We won't take any decision (like early stopping) on this evaluation.

- **globally**, we need both a validation and a test set: a validation set to perform early stopping and a test set to evaluate the model.




Other discussion: how the validation/test set are constructed ?

- local validation set: same distribution as the local training set.
- global validation/test set: same distribution as the original dataset. For instance, consider the following simulation on MNIST 2 partners. Parner A has only the digit 3, Partner B has only the digit 6. The test set should contain the 10 digits and not only the digits 3 and 6.

**Comment by bowni** [comment]
So, status to date @RomainGoussault:

1. Locally:
   - the `Partner` object has attributes for both a val and a test set
   - these are populated by `split_data()`, with an SKL `train_test_split()` once data samples are splitted per partners (so with the same distribution as the partner's data)
   - --> to me we are good
1. Globally:
   - the `Scenario` object has a `Dataset` object attribute which in turn has both a val and a test set
   - the test set is populated at data loading, the val set is populated at `Scenario` initialization with an SKL `train_test_split()` (so with the same distribution as the original dataset)
   - --> to me we are good

Do you see something that need to be improved/fixed?

**Comment by bowni** [comment]
I'll add a note in the `README` to describe all that

**Comment by RomainGoussault** [comment]
Thanks @bowni for the recap!

I thinks we are good.
